rest/tests/base.py

```diff
@@ -0,0 +1,389 @@
+    # Do not mutate these fields!
+    DEFAULT_REPO_DICT = {
+        'name': 'name',
+        'description': 'description',
+    }
+    DEFAULT_VOCAB_DICT = {
+        'name': 'name',
+        'description': 'description',
+        'required': True,
+        'weight': 1000,
+        'vocabulary_type': 'f',
+    }
+    DEFAULT_TERM_DICT = {
+        'label': 'term label',
+        'weight': 1000,
+    }
+
+    def assert_options_head(self, url, expected_status):
+        """Assert OPTIONS and HEAD"""
+        resp = self.client.options(url)
+        self.assertEqual(expected_status, resp.status_code)
+        as_json(resp)
+        self.assertIn("HEAD", resp['ALLOW'])
+        self.assertIn("OPTIONS", resp['ALLOW'])
+        self.assertIn("GET", resp['ALLOW'])
+
+        resp = self.client.head(url)
+        self.assertEqual(expected_status, resp.status_code)
+
+    def get_repositories(self, expected_status=HTTP_200_OK,
+                         skip_options_head_test=False):
+        """Get list of repositories"""
+        url = REPO_BASE
+        if not skip_options_head_test:
+            self.assert_options_head(url, expected_status=expected_status)
+        resp = self.client.get(url)
+        self.assertEqual(expected_status, resp.status_code)
+        if expected_status == HTTP_200_OK:
+            return as_json(resp)
+
+    def create_repository(self, repo_dict=DEFAULT_REPO_DICT,
+                          expected_status=HTTP_201_CREATED,
+                          skip_assert=False):
+        """Helper function to create repository"""
+
+        resp = self.client.post(REPO_BASE, repo_dict)
+        self.assertEqual(expected_status, resp.status_code)
+        if expected_status == HTTP_201_CREATED:
+            result_dict = as_json(resp)
+            if not skip_assert:
+                for key, value in repo_dict.items():
+                    self.assertEqual(value, result_dict[key])
+
+            self.assertIn(reverse(
+                'repository-detail', kwargs={
+                    'repo_slug': result_dict['slug'],
+                }
+            ), resp['Location'])
+
+            return result_dict
+
+    def patch_repository(self, repo_slug, repo_dict,
+                         expected_status=HTTP_200_OK, skip_assert=False):
+        """Update a repository"""
+        resp = self.client.patch(
+            '{repo_base}{repo_slug}/'.format(
+                repo_slug=repo_slug,
+                repo_base=REPO_BASE,
+            ),
+            json.dumps(repo_dict),
+            content_type='application/json',
+        )
+        self.assertEqual(expected_status, resp.status_code)
+        if expected_status == HTTP_200_OK:
+            result_dict = as_json(resp)
+            if not skip_assert:
+                for key, value in repo_dict.items():
+                    self.assertEqual(value, result_dict[key])
+            return result_dict
+
+    def put_repository(self, repo_slug, repo_dict,
```

Comment:
Sorry to be a PITA. I don't see anything in #132 about editing or updating repositories. It looks like #132 is a prerequisite to that. In any case, it looks like there's nothing more to be done before merging this, and the lack of test coverage will always be there to remind us.